MVPInternshipApp.Server/Controllers/CustomerController.cs

```diff
@@ -22,14 +18,15 @@ public CustomerController(SalesprojectContext context)
 
         // GET: api/Customer
         [HttpGet]
-        public async Task<ActionResult<IEnumerable<Customer>>> GetCustomers()
+        public async Task<ActionResult<IEnumerable<CustomerDto>>> GetCustomers()
         {
-            return await _context.Customers.ToListAsync();
+            var customers = await _context.Customers.ToListAsync();
+            return Ok(customers.Select(x => new CustomerDto(x)));
```

Comment:
This looks good **if** we are talking about happy path. What will happen if let's say your controller encounters some unwanted errors? In this case, it is better if you can handle both success and error flow. Can you please implement this in all of your methods not only for Customer Controller? Thanks